### ISSUE TITLE
Failing to Install 

### ISSUE THREAD

**Issue by scalenow** [sjinnovation/CollabAI#19]
Toed following installation instructions and it is failling.

Please help 

(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/client# npm install --legacy-peer-deps
npm warn deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm warn deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
npm warn deprecated workbox-google-analytics@6.6.0: It is not compatible with newer versions of GA starting with v4, as long as you are using GAv3 it should be ok, but the package is not longer being maintained
npm warn deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
npm warn deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm warn deprecated q@1.5.1: You or someone you depend on is using Q, the JavaScript Promise library that gave JavaScript developers strong feelings about promises. They can almost certainly migrate to the native JavaScript promise now. Thank you literally everyone for joining me in this bet against the odds. Be excellent to each other.
npm warn deprecated
npm warn deprecated (For a CapTP with native promises, see @endo/eventual-send and @endo/captp)
npm warn deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
npm warn deprecated abab@2.0.6: Use your platform's native atob() and btoa() methods instead
npm warn deprecated domexception@2.0.1: Use your platform's native DOMException instead
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated @babel/plugin-proposal-optional-chaining@7.21.0: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chaining instead.
npm warn deprecated @babel/plugin-proposal-numeric-separator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separator instead.
npm warn deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nullish-coalescing-operator instead.
npm warn deprecated @babel/plugin-proposal-class-properties@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properties instead.
npm warn deprecated @babel/plugin-proposal-private-methods@7.18.6: This proposal has been merged to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-methods instead.
npm warn deprecated react-google-login@5.2.2: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm warn deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.

added 1805 packages, and audited 1806 packages in 2m

296 packages are looking for funding
  run `npm fund` for details

8 vulnerabilities (2 moderate, 6 high)

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/client# npm audit fix
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: react-google-login@5.2.2
npm error Found: react@18.3.1
npm error node_modules/react
npm error   react@"^18.0.0" from the root project
npm error   peer react@">=16.0.0" from @ant-design/cssinjs@1.22.0
npm error   node_modules/@ant-design/cssinjs
npm error     @ant-design/cssinjs@"^1.21.0" from @ant-design/cssinjs-utils@1.1.1
npm error     node_modules/@ant-design/cssinjs-utils
npm error       @ant-design/cssinjs-utils@"^1.1.1" from antd@5.22.2
npm error       node_modules/antd
npm error         antd@"^5.18.0" from the root project
npm error     @ant-design/cssinjs@"^1.21.1" from antd@5.22.2
npm error     node_modules/antd
npm error       antd@"^5.18.0" from the root project
npm error   73 more (@ant-design/cssinjs-utils, @ant-design/icons, ...)
npm error
npm error Could not resolve dependency:
npm error peer react@"^16 || ^17" from react-google-login@5.2.2
npm error node_modules/react-google-login
npm error   react-google-login@"^5.2.2" from the root project
npm error
npm error Conflicting peer dependency: react@17.0.2
npm error node_modules/react
npm error   peer react@"^16 || ^17" from react-google-login@5.2.2
npm error   node_modules/react-google-login
npm error     react-google-login@"^5.2.2" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /root/.npm/_logs/2024-11-27T02_15_51_850Z-eresolve-report.txt

npm error A complete log of this run can be found in: /root/.npm/_logs/2024-11-27T02_15_51_850Z-debug-0.log
(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/client# cd ../server
(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/server# npm install
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'got@5.7.1',
npm warn EBADENGINE   required: { node: '>=0.10.0 <7' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated babyparse@0.2.1: Deprecated in favour of PapaParse, which now supports Node.js
npm warn deprecated j@0.4.3: The underlying libraries have been unified.  See < https://cdn.sheetjs.com/j/ >.
npm warn deprecated are-we-there-yet@2.0.0: This package is no longer supported.
npm warn deprecated consolidate@0.15.1: Please upgrade to consolidate v1.0.0+ as it has been modernized with several long-awaited fixes implemented. Maintenance is supported by Forward Email at https://forwardemail.net ; follow/watch https://github.com/ladjs/consolidate for updates and release changelog
npm warn deprecated xmldom@0.1.27: Deprecated due to CVE-2021-21366 resolved in 0.5.0
npm warn deprecated gauge@3.0.2: This package is no longer supported.
npm warn deprecated npmlog@5.0.1: This package is no longer supported.
npm warn deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm warn deprecated harb@0.0.7: harb has been merged into xlsx
npm warn deprecated xlsjs@0.7.6: xlsjs has been merged into xlsx
npm warn deprecated xlsx@0.7.12: this version is no longer supported.  More info at https://cdn.sheetjs.com/xlsx/

added 697 packages, and audited 698 packages in 1m

51 packages are looking for funding
  run `npm fund` for details

15 vulnerabilities (1 low, 4 moderate, 9 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/server# npm install
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'got@5.7.1',
npm warn EBADENGINE   required: { node: '>=0.10.0 <7' },
npm warn EBADENGINE   current: { node: 'v18.20.4', npm: '10.7.0' }
npm warn EBADENGINE }

added 41 packages, removed 3 packages, changed 14 packages, and audited 736 packages in 4s

66 packages are looking for funding
  run `npm fund` for details

15 vulnerabilities (1 low, 4 moderate, 9 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
(collabai) root@scalenowai:~/miniconda3/envs/collabai/CollabAI/server# npm start

> collaborativeai-server@1.0.0 start
> set NODE_ENV=local&& nodemon index.js

[nodemon] 3.1.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node index.js`
/root/miniconda3/envs/collabai/CollabAI/server/node_modules/@pinecone-database/pinecone/dist/validator.js:239
            throw new errors_1.PineconeArgumentError(msg);
                  ^

PineconeArgumentError: The client configuration must have required property: apiKey.
    at /root/miniconda3/envs/collabai/CollabAI/server/node_modules/@pinecone-database/pinecone/dist/validator.js:239:19
    at Pinecone._validateConfig (/root/miniconda3/envs/collabai/CollabAI/server/node_modules/@pinecone-database/pinecone/dist/pinecone.js:533:104)
    at new Pinecone (/root/miniconda3/envs/collabai/CollabAI/server/node_modules/@pinecone-database/pinecone/dist/pinecone.js:105:14)
    at file:///root/miniconda3/envs/collabai/CollabAI/server/service/rag.js:5:12
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  cause: undefined
}

Node.js v18.20.4
[nodemon] app crashed - waiting for file changes before starting...
